chemprop/data/splitting.py

```diff
@@ -218,35 +217,45 @@ def _unpack_astartes_result(
     return list(train_idxs), list(val_idxs), list(test_idxs)
 
 
-def split_component(
-    datapointss: Sequence[Sequence[MoleculeDatapoint | MulticomponentDatapoint]],
-    split: SplitType | str = "random",
-    key_index: int = 0,
-    **kwargs,
+def split_data_by_indices(
```

Comment:
`split_data_by_indices()` takes the approach that the format of the data and indices dictates what the return shape should be. There is the simple case that a user gives a list of datapoints and a list of indices and will get back a list of datapoints that correspond to the indices. If the datapoints are a list of lists of datapoints (as is for multicomponent) then lists of lists of datapoints will be returned. Similarly, if the indices are lists of lists (as is for crossfold validation) then a list of lists of datapoints will be returned. It is up to the user to keep track of which case their list of lists of datapoints corresponds to. If the user gives both a list of `n_components` lists of datapoints and a list of `n_indices` lists of indices, they will get back a list of `n_indices` lists of `n_components` lists of datapoints. This is what is used by the CLI for consistency between different cases.